borsh-derive-internal/src/enum_ser.rs

```diff
@@ -109,3 +80,192 @@ pub fn enum_ser(input: &ItemEnum, cratename: Ident) -> syn::Result<TokenStream2>
         }
     })
 }
+
+struct VariantParts {
+    where_predicates: Vec<WherePredicate>,
+    variant_header: TokenStream2,
+    variant_body: TokenStream2,
+    variant_idx_body: TokenStream2,
+}
+fn named_fields(
+    cratename: &Ident,
+    enum_ident: &Ident,
+    variant_ident: &Ident,
+    discriminant_value: &TokenStream2,
+    fields: &FieldsNamed,
+) -> syn::Result<VariantParts> {
+    let mut where_predicates: Vec<WherePredicate> = vec![];
+    let mut variant_header = TokenStream2::new();
+    let mut variant_body = TokenStream2::new();
+    for field in &fields.named {
+        if !contains_skip(&field.attrs) {
+            let field_ident = field.ident.clone().unwrap();
+
+            variant_header.extend(quote! { #field_ident, });
+
+            let field_type = &field.ty;
+            where_predicates.push(
+                syn::parse2(quote! {
+                    #field_type: #cratename::ser::BorshSerialize
+                })
+                .unwrap(),
+            );
+
+            variant_body.extend(quote! {
+                 #cratename::BorshSerialize::serialize(#field_ident, writer)?;
+            })
+        }
+    }
+    // `..` pattern matching works even if all fields were specified
+    let dot_dot = syn::Token![..](Span::mixed_site());
+    variant_header.extend(quote! { #dot_dot });
+    variant_header = quote! { { #variant_header }};
```

Comment:
@dj8yfo Can you help me understand what `Span::mixed_site()` does here? I feel I would write it like this:

```suggestion
    variant_header = quote! { { #variant_header .. }};
```